beacon_node/beacon_chain/src/fork_choice.rs

```diff
@@ -1,13 +1,20 @@
+mod checkpoint_manager;
+
 use crate::{errors::BeaconChainError, metrics, BeaconChain, BeaconChainTypes};
-use lmd_ghost::LmdGhost;
+use checkpoint_manager::{CheckpointManager, CheckpointWithBalances};
 use parking_lot::RwLock;
+use proto_array_fork_choice::ProtoArrayForkChoice;
 use ssz_derive::{Decode, Encode};
-use state_processing::{common::get_attesting_indices, per_slot_processing};
-use std::sync::Arc;
-use store::{BlockRootTree, Error as StoreError, Store};
-use types::{
-    Attestation, BeaconBlock, BeaconState, BeaconStateError, Checkpoint, EthSpec, Hash256, Slot,
-};
+use state_processing::common::get_attesting_indices;
+use std::fs::File;
+use std::io::Write;
+use std::marker::PhantomData;
+use std::time::{SystemTime, UNIX_EPOCH};
+use store::Error as StoreError;
+use types::{Attestation, BeaconBlock, BeaconState, BeaconStateError, Epoch, Hash256};
+
+/// If `true`, fork choice will be dumped to a JSON file in `/tmp` whenever find head fail.
+pub const FORK_CHOICE_DEBUGGING: bool = true;
```

Comment:
We should take this out/set to false before merging

Reply:
I just took it out :) There's now an API endpoint that exports the proto_array as JSON instead.